src/dp_vnf.c

```diff
@@ -38,21 +38,29 @@ int dp_add_vnf(const uint8_t ul_addr6[DP_VNF_IPV6_ADDR_SIZE], enum dp_vnf_type t
 
 	vnf = rte_zmalloc("vnf_handle_mapping", sizeof(struct dp_vnf), RTE_CACHE_LINE_SIZE);
 	if (!vnf) {
-		DPS_LOG_WARNING("VNF handle allocation failed", DP_LOG_VNF_TYPE(type), DP_LOG_VNI(vni),
-						DP_LOG_PORTID(port_id), DP_LOG_IPV4(prefix_ip), DP_LOG_PREFLEN(prefix_len));
+		if (pfx_ip->ip_type == RTE_ETHER_TYPE_IPV4)
+			DPS_LOG_WARNING("VNF handle allocation failed", DP_LOG_VNF_TYPE(type), DP_LOG_VNI(vni),
+							DP_LOG_PORTID(port_id), DP_LOG_IPV4(pfx_ip->ipv4), DP_LOG_PREFLEN(prefix_len));
+		else
+			DPS_LOG_WARNING("VNF handle allocation failed", DP_LOG_VNF_TYPE(type), DP_LOG_VNI(vni),
+							DP_LOG_PORTID(port_id), DP_LOG_IPV6(pfx_ip->ipv6), DP_LOG_PREFLEN(prefix_len));
```

Comment:
I think we could create a `DP_LOG_IP()` that uses a ternary for `ip->ip_type` to encapsulate  logging for `struct dp_ip_address` It will simplify code in multiple places in this PR.

Reply:
You mean smt like this:

> #define DP_LOG_IP(ip_addr) \
       ((ip_addr)->ip_type == RTE_ETHER_TYPE_IPV4 ? \
               DP_LOG_IPV4((ip_addr)->ipv4) : \
               DP_LOG_IPV6((ip_addr)->ipv6))

But ternary would return integer for the ipv4 case and pointer for the other case. I would leave it as it is and we can improve the logging as a refactoring

Reply:
Oh you're right, the ternary is runtime, but the macro is compile-time oriented, sorry.
you need to actually define a new type:
```
#define _DP_LOG_IPVx(KEY, VALUE) KEY, (VALUE)->ip_type == RTE_ETHER_TYPE_IPV4 ? _DP_LOG_FMT_IPV4 : _DP_LOG_FMT_IPV6, VALUE
```
And then the macro itself
```
#define DP_LOG_IP(VALUE) _DP_LOG_IP((VALUE)->ip_type == RTE_ETHER_TYPE_IPV4 ? "ipv4" : "ipv6", (VALUE)->ip_type == RTE_ETHER_TYPE_IPV4 ? (VALUE)->ipv4 : (VALUE)->ipv6)
```
But yea, I can just note it down for later commit in logging.